dash/development/component_generator.py

```diff
@@ -75,87 +85,118 @@ def generate_components(components_source, project_shortname,
+        "using react-docgen. Then map the metadata to python classes.",
     )
-    parser.add_argument('components_source',
-                        help='React components source directory.')
+    parser.add_argument("components_source",
+                        help="React components source directory.")
     parser.add_argument(
-        'project_shortname',
-        help='Name of the project to export the classes files.'
+        "project_shortname",
+        help="Name of the project to export the classes files."
     )
     parser.add_argument(
-        '-p', '--package-info-filename',
-        default='package.json',
-        help='The filename of the copied `package.json` to `project_shortname`'
+        "-p",
+        "--package-info-filename",
+        default="package.json",
+        help="The filename of the copied `package.json` \
+to `project_shortname`",
     )
     parser.add_argument(
-        '-i', '--ignore',
-        default='^_',
-        help='Files/directories matching the pattern will be ignored'
+        "-i",
+        "--ignore",
+        default="^_",
+        help="Files/directories matching the pattern will be ignored",
     )
     parser.add_argument(
-        '--r-prefix',
-        help='Experimental: specify a prefix for DashR component names, write'
-             'DashR components to R dir, create R package.'
+        "--r-prefix",
+        help="Specify a prefix for DashR component names, write "
+        "DashR components to R dir, create R package.",
+    )
+    parser.add_argument(
+        "--r-depends",
+        default="",
+        help="Specify a comma-separated list of R packages to be "
+        "inserted into the Depends field of the DESCRIPTION file.",
+    )
+    parser.add_argument(
+        "--r-imports",
+        default="",
+        help="Specify a comma-separated list of R packages to be "
+        "inserted into the Imports field of the DESCRIPTION file.",
+    )
+    parser.add_argument(
+        "--r-suggests",
+        default="",
+        help="Specify a comma-separated list of R packages to be "
+        "inserted into the Suggests field of the DESCRIPTION file.",
     )
 
     args = parser.parse_args()
     generate_components(
-        args.components_source, args.project_shortname,
+        args.components_source,
+        args.project_shortname,
         package_info_filename=args.package_info_filename,
         ignore=args.ignore,
-        rprefix=args.r_prefix)
+        rprefix=args.r_prefix,
+        rdepends=args.r_depends,
+        rimports=args.r_imports,
+        rsuggests=args.r_suggests,
+    )
 
 
 # pylint: disable=undefined-variable
 def byteify(input_object):
     if isinstance(input_object, dict):
-        return OrderedDict([
-            (byteify(key), byteify(value))
-            for key, value in input_object.iteritems()
-        ])
+        return OrderedDict(
+            [(byteify(key), byteify(value)) for key,
+             value in input_object.iteritems()]
```

Comment:
huh, was it black's choice to break between `key, value`? Kind of a subtle case, but I'd say it's objectively wrong - ie more confusing - that way. Maybe it would make a better choice if parentheses were put around `(key, value)`?

Totally non-blocking, I'm just trying to wrap my head around what black does as it seems the push to use it more places is inevitable...

Reply:
Yeah, to be entirely honest, so far I'm not yet a fan of `black`. To be fair, `pylint` and `flake8` took a little getting used to as well, so I'm giving it time.

For one thing, it seems that the reformatting occasionally requires another round of linting after running `black`, which is frustrating if everything seemed OK beforehand. 

I don't like the "new" formatting ☝️(in this case) at all. I thought it was just me!

Reply:
had some experience using black with numpy heavy code, some of the formatting result is ugly, we just block the unhappy chunk with # fmt: on/off. but the most attractive point for me is no more argument :)

Reply:
Right, no more style arguments is worth a LOT 💰 - probably even worth a little bit of reduced readability

Reply:
Yep, I completely agree.

A major role of guidelines is to force consensus where it was not easily achieved otherwise. It doesn't mean the consensus is optimal or even objectively correct, but often the common ground offered by a third party is more attractive (particularly in the face of argument).

Reply:
FYI I will also apply `black` for pytest fixture porting